acceptance/setup/common/pre-suite/031_add_local_nix_user.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'bolt_setup_helper'
+
+test_name "Configure a bolt users on non-windows" do
+  extend Beaker::HostPrebuiltSteps
+  extend Acceptance::BoltSetupHelper
+
+  skip_test('Not applicable windows') if bolt['platform'] =~ /windows/
+  step "Create a non-root user to test local transport run-as" do
+    # Use Bolt's puppet for package testing
+    result = on(bolt, 'stat /usr/local/bin/puppet', acceptable_exit_codes: [0, 1])
+    puppet_path = result.exit_code == 0 ? '/usr/local/bin/puppet' : '/opt/puppetlabs/bolt/bin/puppet'
+
+    case bolt['platform']
+    when /osx/
+      dir = bolt.tmpdir('osx_manifest')
+      # Use beaker to set osx user and ensure they have a homedir
+      bolt.user_present(local_user)
+      create_remote_file(bolt, "#{dir}/osx_user.pp", <<~MANIFEST)
+      exec { "/usr/sbin/createhomedir -c -l -u #{local_user}":
```

Comment:
I guess 2 things:
1. It seems like rather than having a puppet exec we can just execute this on the system with beaker? OR use the directory resource to create the homedir?
2. I thought `resource user managehome=` would take care of this, but osx might be an edge case or I might be missing something? I'll try it out after lunch

Reply:
`managehome` for puppet user resource definitely does not work out of the box with osx. I guess the alternative to an exec would be something like:
```ruby
on(bolt, "[[ ! -d /Users/#{local_user}/Library ]] && /usr/sbin/createhomedir -c -l -u #{local_user}", acceptable_exit_codes: [0, 1])
```
Which seems bad. I guess I could copy up a shell script, or not rely on that shell shortcut (which produces the 1 exit code).

Reply:
In that case maybe leave a comment explaining that. Something like "managehome isn't reliable for OSX, and using Puppet is easier than managing user creation logic with bash" or something to that effect (affect? pretty sure it's effect).